torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -522,6 +522,7 @@ def __init__(
         # implemented using post-save and pre-load hooks
         _init_state_dict_state(self)
         _register_all_state_dict_hooks(self)
+        self.zero = torch.tensor(0.0, device=self.compute_device)
```

Comment:
Should we initialize this lazily (upon calling `clip_grad_norm_` and actually needing to use this)?

Maybe we can put it in a private attribute like `self._zero_scalar`?